modules/platforms/dotnet/Apache.Ignite.Core/Impl/Services/ServiceProxyInvoker.cs

```diff
@@ -102,6 +102,12 @@ internal static class ServiceProxyInvoker
                         methodName, svcType, argsLength));
 
             // 2) There is more than 1 method with specified name - resolve with argument types.
+            methods = methods.Where(m => AreMethodArgsCompatible(arguments, m.GetParameters(), true)).ToArray();
```

Comment:
I tried reverting all changes in this file, and all tests pass. Please double check - either we need more tests, or those changes are unnecessary.

Reply:
`ServicesClientTestBinaryArrays#testOverloadResolution` failed without changes with exception: 
> Apache.Ignite.Core.Client.IgniteClientException : Failed to invoke proxy: there are 2 methods 'Foo' in type 
> 'Apache.Ignite.Core.Tests.Client.Services.TestServiceOverloads' with (Person[]) arguments, can't resolve ambiguity.

Reply:
Now I understand, we attempt a more strict check to reduce ambiguity. 

Can you please rename `checkEq` to `strictTypeCheck` and invert the logic? Right now we perform equality check when `checkEq = false`, which is confusing.

Reply:
And let's add a condition to make it clear that we try to resolve ambiguity here:

```suggestion
        if (methods.Length > 1)
            methods = methods.Where(m => AreMethodArgsCompatible(arguments, m.GetParameters(), true)).ToArray();
```

Reply:
Fixed.